core/src/main/scala/org/apache/spark/executor/ShuffleWriteMetrics.scala

```diff
@@ -17,40 +17,65 @@
 
 package org.apache.spark.executor
 
+import org.apache.spark.{Accumulator, InternalAccumulator}
 import org.apache.spark.annotation.DeveloperApi
 
 
 /**
  * :: DeveloperApi ::
- * Metrics pertaining to shuffle data written in a given task.
+ * A collection of accumulators that represent metrics about writing shuffle data.
```

Comment:
It just occurred to me that maybe we should add a note here and in `ShuffleReadMetrics` to explain that instances of this class are not thread-safe.